### PR TITLE
Zoomable: add option to pan vertically

### PR DESCRIPTION
**Pull request by jordalgo**
Add 'vertical_pan_direction' which enables vertical panning on Zoomable
instances. At the moment this only works reliably on the Flamechart.

### CODE REVIEW

File: js/c3-plot.coffee [hunk 1]
```diff
@@ -363,11 +363,14 @@ class c3.Plot.Zoomable extends c3.Plot
     # [Number, String] The maximum zoom factor the user is allowed to zoom in
     # If set to _integer_ then allow zooming only until pixels are integer values.
     zoom_extent: undefined
+    # [String] Enables vertical panning. Values are either 'up' or 'down'. At the moment this only works reliably on the Flamechart
+    vertical_pan_direction: undefined
```
Comment (drarmstr) [inline js/c3-plot.coffee]: why do we need `up` or `down` here?  Can we key that off `v_orient`?

File: js/c3-plot.coffee [hunk 2]
```diff
@@ -470,6 +482,17 @@ class c3.Plot.Zoomable extends c3.Plot
             @trigger 'restyle', true
         return if domain[0]==extent[0] and domain[1]==extent[1] then null else new_domain
 
+     pan: (dy) =>
```
Comment (drarmstr) [inline js/c3-plot.coffee:487]: Do you expect to persist the vertical panning state at the app level?

Comment (drarmstr) [reply js/c3-plot.coffee]: Since we can also pan horizontally, then to maintain external state we would also need to pass in `dx`

Comment (drarmstr) [reply js/c3-plot.coffee:487]: Add a comment that `dx` would be handled by the `extent` in `focus`

**Comment by jordalgo** [comment]
@drarmstr I think this is ready for another look.

**Review by drarmstr — APPROVED** [review]
Thanks for adding this!!

**Pull request merged**